ndb/src/google/cloud/ndb/tasklets.py

```diff
@@ -366,6 +380,39 @@ def tasklet_wrapper(*args, **kwargs):
     return tasklet_wrapper
 
 
+def wait_any(futures):
+    """Wait for any of several futures to finish.
+
+    Args:
+        futures (Sequence[Future]): The futures to wait on.
+
+    Returns:
+        Future: The first future to be found to have finished.
+    """
+    if not futures:
+        return None
+
+    while True:
+        for future in futures:
+            if future.done():
+                return future
+
+        _eventloop.run1()
```

Comment:
Just for my own edification, can you explain what this line does?

Reply:
Sure. I added #7068 in attempt to make code like this easier to understand. Possibly more can still be done.

Essentially, the `run1()` call does one thing on the eventloop—if there's a callback scheduled on the eventloop it calls that. If there aren't any call backs scheduled but there are pending gRPC calls, it waits for a gRPC call to finish and then runs its callback. Basically, this is the function that causes "something" to happen on the event loop where eventually one of those things will cause one of the futures we're waiting on to finish, and then we'll return that future.

I think the most confusing thing about the event loop is that it's not really an event loop in the way we typically think about it. Generally, if we're writing event driven code, everything we write is an event handler and the event loop is always running and calling into our event handlers. In NDB, on the other hand, the event loop is really just a task queue that we poke (by calling `run1()`) whenever we want something out of it. So, the flow of control is inverted, really, if you think about it. Maybe it shouldn't be called an "event loop". 

The one and only time the event loop actually loops is when we call `run()`, which we only do when we're wrapping up our context, to make sure anything in the queue gets run before exiting the context. You can imagine someone might have fired off a `put_async` without waiting for a response, but even if the calling code doesn't need the result, we still have to make sure it gets run. 

Does this, together with the comments in #7068, help it start to  make sense?